packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3472,29 +3472,21 @@ export class ContainerRuntime
 		return result;
 	}
 
-	detachHead(): { merge(): void; pause(): Promise<void>; dispose(): void } {
+	enterStagingMode = (): { commitChanges(): void; discardChanges(): void } => {
```

Comment:
```suggestion
	public readonly enterStagingMode = (): { readonly commitChanges: () => void; readonly discardChanges: () => void } => {
```

or

```ts
	// elsewhere...
	export type StageControls = { readonly commitChanges: () => void; readonly discardChanges: () => void };
	// meanwhile...
	public readonly enterStagingMode = (): StageControls => {
```

Reply:
I should note the latter is preferable, as surely the customer will be holding onto these controls somewhere ([e.g. like this](https://github.com/ChumpChief/FluidFramework/compare/test/detached-head-branching...ChumpChief:FluidFramework:IntegrateDetachedHeadPrototype?expand=1#diff-e1255246a07a5e3a0efe60c0fde9dc0d8f1347fccf21331343529690125fdebdR19)) and will appreciate having a nice type for them.